rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -782,6 +1034,172 @@ mod tests {
         .unwrap();
     }
 
+    #[test]
+    fn test_cast_date32_to_date64() {
+        let a = Date32Array::from(vec![10000, 17890]);
+        assert_eq!(
```

Comment:
It's best if we can avoid using string to compare.

Reply:
Done, removed the string comparisons